src/OpenTK.Windowing.Desktop/GLFWProvider.cs

```diff
@@ -28,19 +30,71 @@ internal static class GLFWProvider
 
         /// <summary>
         /// Gets a value indicating whether the <see cref="Thread.CurrentThread"/> is the same as the GLFW main thread.
+        /// If <see cref="CheckForMainThread"/> is false this function will always return true.
         /// </summary>
-        public static bool IsOnMainThread => _mainThread == Thread.CurrentThread;
+        public static bool IsOnMainThread
+        {
+            get
+            {
+                if (CheckForMainThread == false)
+                {
+                    return true;
+                }
+                else
+                {
+                    return _mainThread?.ManagedThreadId == Thread.CurrentThread.ManagedThreadId;
+                }
+            }
+        }
```

Comment:
```suggestion
        public static bool IsOnMainThread => !CheckForMainThread || _mainThread == Thread.CurrentThread;
```
All I got

Reply:
I changed `!CheckForMainThread` to `CheckForMainThread == false` as I find that pattern less easy to miss.